packages/office-ui-fabric-react/src/components/Callout/CalloutContent.tsx

```diff
@@ -76,7 +76,8 @@ export class CalloutContent extends BaseComponent<ICalloutProps, ICalloutState>
   }
 
   public componentWillUpdate(newProps: ICalloutProps) {
-    if (newProps.targetElement !== this.props.targetElement || newProps.target !== this.props.target) {
+    // If the target element changed, find the new one. If we are tracking target with class name, always find element because we do not know if fabric has rendered a new element and disposed the old element.
+    if (newProps.targetElement !== this.props.targetElement || newProps.targetElement == null) {
```

Comment:
I see what you mean now. That change makes sense, but I don't believe your fix will achieve what you want. What you're saying, and it make some amount of sense, is that target should be recalculated every time and that the if statement should really just be removed.